src/lib/plugin.js

```diff
@@ -235,76 +235,85 @@ class FiveBellsLedger extends EventEmitter2 {
       return new WebSocket(notificationsUrl)
     })
 
-    return new Promise((resolve, reject) => {
-      this.connection = reconnect({immediate: true}, (ws) => {
-        ws.on('open', () => {
-          if (!this.connected) {
-            this.emit('connect')
-            this.connected = true
-          }
-          debug('ws connected to ' + notificationsUrl)
-        })
-        ws.on('message', (rpcMessageString) => {
-          let rpcMessage
-          try {
-            rpcMessage = JSON.parse(rpcMessageString)
-          } catch (err) {
-            debug('invalid notification', rpcMessageString)
-            return
-          }
-
-          if (rpcMessage.method === 'connect') return resolve(null)
-          co.wrap(this._handleIncomingRpcMessage)
-            .call(this, rpcMessage)
-            .then(() => {
-              if (this.debugReplyNotifications) {
-                ws.send(JSON.stringify({ result: 'processed' }))
-              }
-            })
-            .catch((err) => {
-              debug('failure while processing notification: ' +
-                (err && err.stack) ? err.stack : err)
-              if (this.debugReplyNotifications) {
-                ws.send(JSON.stringify({
-                  result: 'ignored',
-                  ignoreReason: {
-                    id: err.name,
-                    message: err.message
-                  }
-                }))
+    return Promise.race([
```

Comment:
Can you add some comments to this code block to describe what the right behavior is?